src/ServiceInsight/Framework/Settings/ApplicationConfiguration.cs

```diff
@@ -0,0 +1,27 @@
+﻿namespace ServiceInsight.Framework.Settings
+{
+    using System;
+    using System.Configuration;
+
+    public class ApplicationConfiguration
+    {
+        public static bool SkipCertificateValidation { get; private set; }
+
+        public static void Initialize()
+        {
+            SkipCertificateValidation = GetValue<bool>("SkipCertificateValidation");
+        }
+
+        private static T GetValue<T>(string key)
+        {
+            try
+            {
+                return (T)Convert.ChangeType(ConfigurationManager.AppSettings[key], typeof(T));
+            }
+            catch (Exception ex)
+            {
+                throw new ApplicationException($"The value of '{key}' in app.config is not a valid {typeof(T).Name} type.", ex);
```

Comment:
I see a few problems with this exception message:

- The value of a given app setting is not a "type", but rather a string representation of an instance of a type.
- The user will never see a file named "app.config".
- The value may be missing entirely, as another reason for the conversion to fail.
- We should really use the full type name.

Perhaps:

```C#
$"Application setting '{key}' is either missing or cannot be converted to type {typeof(T)}."
```

However, there is also another problem here. If `T` is a nullable type, e.g. `string`, the behaviour will be different to that of a non-nullable type, e.g. `bool`. In the first case, a null will be returned. In the latter, an exception will be thrown. Perhaps we should first retrieve the value and check if it is null, and throw a separate exception for that, and only if it is not null, should we then proceed to the conversion.

Reply:
I'd say over-thinking it to solve all possible scenarios would be premature optimization. It has taken us quite a few years without needing the app.config file at all. I was thinking to go with a more specific/simpler version to target the problem at hand (e.g. a non-generic method that deals with a bool).

Reply:
BTW, the problem of having a typed access to app.config was already solved by Settings files, aren't we re-creating the wheel just for consistency sake?